Engine/Module/ModuleInput.cpp

```diff
@@ -117,6 +153,47 @@ update_status ModuleInput::PreUpdate()
 			mouse_bible[(MouseButton)event.button.button] = KeyState::UP;
 			break;
 
+		case SDL_CONTROLLERBUTTONDOWN:
+			//TODO how to handle multiple controllers
+			controller_bible[(ControllerCode)event.cbutton.button] = KeyState::DOWN;
+			break;
+
+		case SDL_CONTROLLERBUTTONUP:
+			controller_bible[(ControllerCode)event.cbutton.button] = KeyState::UP;
+			break;
+		
+		case SDL_CONTROLLERAXISMOTION:
```

Comment:
This case is pretty big, maybe it is better to move it into a different function.

Reply:
Okay I will move it to a separate function thx :D